rust/lancedb/src/embeddings/bedrock.rs

```diff
@@ -0,0 +1,210 @@
+            .len(len)
+            .add_child_data(inner.into_data())
+            .build()?;
+
+        Ok(Arc::new(FixedSizeListArray::from(array_data)))
+    }
+
+    fn compute_query_embeddings(&self, input: Arc<dyn Array>) -> Result<Arc<dyn Array>> {
+        let arr = self.compute_inner(input)?;
+        Ok(Arc::new(arr))
+    }
+}
+
+impl std::fmt::Debug for BedrockEmbeddingFunction {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BedrockEmbeddingFunction")
+            .field("model", &self.model)
+            // Skip client field as it doesn't implement Debug
+            .finish()
+    }
+}
+
+impl BedrockEmbeddingFunction {
+    fn compute_inner(&self, source: Arc<dyn Array>) -> Result<Float32Array> {
+        if source.is_nullable() {
+            return Err(Error::InvalidInput {
+                message: "Expected non-nullable data type".to_string(),
+            });
+        }
+
+        if !matches!(source.data_type(), DataType::Utf8 | DataType::LargeUtf8) {
+            return Err(Error::InvalidInput {
+                message: "Expected Utf8 data type".to_string(),
+            });
+        }
+
+        let mut builder = Float32Builder::new();
+
+        let texts = match source.data_type() {
+            DataType::Utf8 => source
+                .as_string::<i32>()
+                .into_iter()
+                .map(|s| s.expect("array is non-nullable").to_string())
+                .collect::<Vec<String>>(),
+            DataType::LargeUtf8 => source
+                .as_string::<i64>()
+                .into_iter()
+                .map(|s| s.expect("array is non-nullable").to_string())
+                .collect::<Vec<String>>(),
+            _ => unreachable!(),
+        };
+
+        for text in texts {
+            let request_body = match self.model {
+                BedrockEmbeddingModel::TitanEmbedding => {
+                    json!({
+                        "inputText": text
+                    })
+                }
+                BedrockEmbeddingModel::CohereLarge => {
+                    json!({
+                        "texts": [text],
+                        "input_type": "search_document"
+                    })
+                }
+            };
+
+            let client = self.client.clone();
+            let model_id = self.model.model_id().to_string();
+            let request_body = request_body.clone();
+
+            let response = block_in_place(move || {
+                Handle::current().block_on(async move {
+                    client
+                        .invoke_model()
+                        .model_id(model_id)
+                        .body(aws_sdk_bedrockruntime::primitives::Blob::new(
+                            serde_json::to_vec(&request_body).unwrap(),
+                        ))
+                        .send()
+                        .await
+                })
+            })
+            .unwrap();
```

Comment:
I'm not sure if there are complexities with rate limiting you would have to handle, but it might be worth it to run the queries concurrently. Right now it's entirely serial, which could be slow.